connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -544,6 +555,95 @@ public int generation() {
         return generation;
     }
 
+    @Override
+    public synchronized void getConfigDef(final String connType, final Callback<ConfigDefInfo> callback) {
+        log.trace("Submitting get connector configuration definitions request {}", connType);
+
+        addRequest(
+            new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    ConfigDef configDef = getConfig(connType);
+                    Collection<ConfigDef.ConfigKey> configKeys = configDef.configKeys().values();
+                    List<ConfigKeyInfo> configKeyInfos = new LinkedList<>();
+                    for (ConfigDef.ConfigKey configKey: configKeys) {
+                        configKeyInfos.add(convertConfigKey(configKey));
+                    }
+                    ConfigDefInfo configDefInfo = new ConfigDefInfo(connType, configKeyInfos);
+                    callback.onCompletion(null, configDefInfo);
+                    return null;
+                }
+            },
+            forwardErrorCallback(callback)
+        );
+    }
+
+    @Override
+    public synchronized void validateConfigs(final String connType, final Map<String, String> connectorConfig, final Callback<ConfigInfos> callback) {
+        log.trace("Submitting validate configuration request {}", connType);
+
+        addRequest(
```

Comment:
Same question here. (If we want to validate against shared state like connector names, then we will need this).